### DISCUSSION TITLE
Use global neovim installation

### DISCUSSION THREAD

**Discussion by shahamran** [amitds1997/remote-nvim.nvim#117] [Q&A] [answered]
Hi @amitds1997, thanks for this plugin, I'm looking forward to be able to use it!
My (company's) remote host has an unsupported OS (ubuntu 18.04). How do I make remote-nvim use the globally installed nvim on remote, instead of trying to install its own version?

**Comment by shahamran** [comment]
To clarify, when prompted for version selection, I used "Nightly version (locally installed)", but in the "Installing Neovim" step, this was printed:

```
Neovim installation is corrupted. Would re-install.
```

**Reply by amitds1997** [reply]
I suppose that line could be improved. It just means that it was not able to execute `nvim --version` on the remote for some reason. This means that it would also be printed the first time the plugin executes when it is supposed to create that path. Does it not install after that?

**Reply by amitds1997** [reply]
Currently, the plugin does have an option to use the system Neovim, but it is only shown when the plugin determines that binary for the OS is not available (e.g. FreeBSD, etc.) and the remote system has a Neovim binary on the path. I guess we can provide this option universally in the same select menu as the rest of the version selection.

**Reply by shahamran** [reply]
Thanks for the quick reply!
The issue with ubuntu18 is that it uses an old version of glibc, so while technically there is an official build for this platform, in practice you need to download an unofficial build (or build from source).
As I see it, any (or all) of the options you listed will be helpful:
* Add option to use globally installed neovim on remote, even if a binary for that platform exists.
* Add option to build from source.
* Detect glibc version and use the unofficial build (only for nightly): <https://github.com/neovim/neovim-releases/releases>.

**Reply by shahamran** [reply]
I think adding a config option for passing arguments to the remote `neovim_install.sh` will solve this, no?
Then I can just set `-m source` or `-m system`.

**Reply by amitds1997** [reply, marked as answer]
Thanks for doing the relevant research. Allowing that configurability would definitely help. But this would be cumbersome to do right now, as the plugin configuration is not host specific and I doubt you want to build from source for all your remote machines and that would make it a hassle to go change it in your config every time. 

For now, I am thinking simplifying the selection menu to always provide the user with option to use system wide Neovim installation can help. Later on, I hope to build a single form using something like https://nui-components.grapp.dev/docs/components/form to create a single form that would simplify the configuration quite a bit and make it easy for me to maintain as well.

Let me know what you think about this "workaround" for now, and I would send out a PR tom to add this.

**Reply by shahamran** [reply]
Always providing the "use system neovim" option would be great, thanks!

**Reply by amitds1997** [reply]
Starting [v0.3.3](https://github.com/amitds1997/remote-nvim.nvim/releases/tag/v0.3.3), you should have the option to use the remote Neovim installation, if available.
